hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -284,17 +284,22 @@ public void forEach(BiConsumer<Data, Record> consumer,
                         boolean backup, boolean includeExpiredRecords) {
 
         long now = getNow();
-        Iterator<Map.Entry<Data, Record>> entries = storage.mutationTolerantIterator();
-        while (entries.hasNext()) {
-            Map.Entry<Data, Record> entry = entries.next();
+        storage.beforeOperation();
```

Comment:
I think all iterator accesses are instances of map operation. So `storage.beforeOperation()` would be called in an upper layer, no?

Reply:
Unfortunately in some cases, we call it without `MapOperation` or any other `Operation` context.